common/flatpak-dir.c

```diff
@@ -326,18 +345,18 @@ flatpak_remote_state_unref (FlatpakRemoteState *remote_state)
 
 void
 flatpak_remote_state_add_sideload_repo (FlatpakRemoteState *self,
-                                        const char          *path)
+                                        GFile *dir)
 {
-  g_autoptr(GFile) f = g_file_new_for_path (path);
-  g_autoptr(GFile) summary_path = g_file_get_child (f, "summary");
+  g_autoptr(GFile) summary_path = NULL;
   g_autoptr(GMappedFile) mfile = NULL;
-  g_autoptr(OstreeRepo) sideload_repo = ostree_repo_new (f);
+  g_autoptr(OstreeRepo) sideload_repo = NULL;
 
   /* Sideloading only works if collection id is set */
   if (self->collection_id == NULL)
     return;
 
-  /* TODO: The sideloadstates are duplicated for each remote with the collection id set, we should maybe reuse them */
```

Comment:
It is still true, but I don't think its worth fixing, because doing so would break the abstractions a lot (share stuff from different remotes in one FlatpakRemote) while not really saving a lot of resources (we're mmaping the same file readonly multiple times, should be *essentially* free).